syft/serde.py

```diff
@@ -496,111 +498,61 @@ def _detail_ellipsis(ellipsis: bytes) -> Ellipsis:
     return ...
 
 
-# PointerTensor
-
-
-def _simplify_pointer_tensor(ptr: PointerTensor) -> Dict:
+def _simplify_pointer_tensor(ptr: PointerTensor) -> tuple:
     """
     This function takes the attributes of a PointerTensor and saves them in a dictionary
-
     Args:
         PointerTensor: a PointerTensor
-
     Returns:
         Dict: a dictionary holding the attributes of the PointerTensor
-
     Usage:
         data = _simplify_pointer_tensor(ptr)
-
     """
 
-    data = vars(ptr).copy()
-    for k, v in data.items():
+    return (ptr.id, ptr.id_at_location, ptr.location.id)
 
-        if isinstance(v, AbstractWorker):
-            data[k] = v.id
+    # a more general but slower/more verbose option
 
-    return _simplify_dictionary(data)
+    # data = vars(ptr).copy()
+    # for k, v in data.items():
+    #     if isinstance(v, AbstractWorker):
+    #         data[k] = v.id
+    # return _simplify_dictionary(data)
 
 
-def _detail_pointer_tensor(data: Dict) -> PointerTensor:
+def _detail_pointer_tensor(data: tuple) -> PointerTensor:
     """
     This function reconstructs a PointerTensor given it's attributes in form of a dictionary.
     We use the spread operator to pass the dict data as arguments
     to the init method of PointerTensor
-
     Args:
         Dict: a dictionary holding the attributes of the PointerTensor
-
     Returns:
         PointerTensor: a PointerTensor
-
     Usage:
         ptr = _detail_pointer_tensor(data)
-
     """
-    new_data = {}
-    for k, v in data.items():
-
-        key = k.decode()
-        if type(v) is bytes:
-            val_str = v.decode()
-            try:
-                val = syft.local_worker.get_worker(val_str, fail_hard=True)
-            except WorkerNotFoundException:
-                val = val_str
-        else:
-            val = v
-        new_data[key] = val
-
-    return PointerTensor(**new_data)
-
-
-# High Level Simplification Router
-
-
-# PointerTensor
+    # TODO: fix comment for this and simplifier
+    print(data)
 
+    return PointerTensor(
+        id=data[0],
+        id_at_location=data[1],
+        location=syft.torch.hook.local_worker.get_worker(data[2]),
+    )
 
-def _simplify_pointer_tensor(ptr: PointerTensor) -> Dict:
-    """
-    This function takes the attributes of a PointerTensor and saves them in a dictionary
-    Args:
-        PointerTensor: a PointerTensor
-    Returns:
-        Dict: a dictionary holding the attributes of the PointerTensor
-    Usage:
```

Comment:
Usage apparently is not supported by sphinx/napoleon (https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html). Usage should be replaced by Example as shown in this example (https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google).

Reply:
Changed!